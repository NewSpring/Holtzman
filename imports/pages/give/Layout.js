// @flow
import Split, { Left, Right } from "../../blocks/split";
import Dashboard from "../../components/dashboard/";

const subNav = [
  {
    isActive: true,
    linkUrl: "https://my.newspring.cc/give",
    onClick: () => {},
    title: "Home",
  },
  {
    isActive: false,
    linkUrl: "https://my.newspring.cc/give/now",
    onClick: () => {},
    title: "Give Now",
  },
  {
    isActive: false,
    linkUrl: "https://my.newspring.cc/give/history",
    onClick: () => {},
    title: "History",
  },
];

const Layout = () => (
  <div>
    <Split>
      <Right
        background="//s3.amazonaws.com/ns.images/newspring/christmasoffering/christmas_offering_1x1.png"
      />
    </Split>
    <Left>
      <Dashboard
        title="Your Giving"
        subNav={subNav}
      >
        <div>Left Side</div>
      </Dashboard>
    </Left>
  </div>
);

export default Layout;
