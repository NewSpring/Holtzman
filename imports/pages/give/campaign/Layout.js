// @flow
/* eslint-disable react/no-danger */
// $FlowMeteor
import { find } from "ramda";
import { Link } from "react-router";

import Split, { Left, Right } from "../../../components/@primitives/layout/split";
import Meta from "../../../components/shared/meta";
import AddToCart from "../../../components/giving/add-to-cart";

type ILayout = {
  account: Object,
};

const Layout = ({ account }: ILayout) => (
  <div>
    <Split nav classes={["background--light-primary"]}>
      <Meta
        title={account.name}
        description={account.summary}
        image={
          account.images ? find(account.images, { fileLabel: "2:1" }).cloudfront : account.image
        }
        meta={[
          { property: "og:type", content: "article" },
        ]}
      />

      <Right
        background={
          account.images ?
            find(account.images, { fileLabel: "2:1" }).cloudfront :
            account.image
        }
        mobile
      />
      <Right
        background={
          account.images ?
            find(account.images, { fileLabel: "1:2" }).cloudfront :
            account.image
        }
        mobile={false}
      />

    </Split>
    <Left scroll classes={["background--light-primary"]}>
      <Link
        to="/give/now"
        className={
          "locked-top locked-left soft-double@lap-and-up soft " +
          "h7 text-dark-secondary plain visuallyhidden@handheld"
        }
      >
        <i
          className="icon-arrow-back soft-half-right display-inline-block"
          style={{ verticalAlign: "middle" }}
        />
        <span
          className="display-inline-block"
          style={{ verticalAlign: "middle", marginTop: "5px" }}
        >
          Back
        </span>
      </Link>
      <div className="soft@lap-and-up soft-double-top@lap-and-up">
        <div className="soft soft-double-bottom soft-double-top@lap-and-up">

          <h2>{account.name}</h2>
          <div dangerouslySetInnerHTML={{ __html: account.description }} />
        </div>
      </div>

      <div className="background--light-secondary">
        <div className="constrain-copy soft-double@lap-and-up">
          <div className="soft soft-double-bottom soft-double-top@lap-and-up">
            <AddToCart accounts={[account]} donate />
          </div>
        </div>
      </div>

    </Left>
  </div>
);

export default Layout;
