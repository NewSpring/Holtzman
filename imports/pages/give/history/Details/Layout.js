/* eslint-disable react/no-danger */
import { Component, PropTypes } from "react";
import moment from "moment";
import { Link } from "react-router";

import Split, { Left, Right } from "../../../../blocks/split";

import { Spinner } from "../../../../components/loading";
import Currency from "../../../../components/currency";
import SideBySide from "../../../../components/cards/SideBySide";
import Meta from "../../../../components/meta";
import AccountType from "../../../../components/accountType";

export default class Layout extends Component {

  static propTypes = {
    loadingEntries: PropTypes.bool,
    entries: PropTypes.array,
    transaction: PropTypes.object,
  }

  formatDate = (date) => (
    moment(new Date(date)).format("MMM D, YYYY")
  )

  /* eslint-disable max-len */
  render() {
    const { loadingEntries, entries } = this.props;
    return (
      <div>
        <Split nav classes={["background--light-primary"]}>
          <Meta title="Giving History" />
          <Right
            background="//dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/all/heroes/newspring/campuses/Florence.1.2x1_1700_850_90_c1.jpg"
            mobile={false}
          />
        </Split>
        <Left scroll ref="container" classes={["background--light-secondary"]}>
          <div
            className={
              "soft-double-sides@lap-and-up soft-double-ends@lap-and-up " +
              "soft background--light-primary"
            }
          >
            {(() => {
              if (process.env.WEB) {
                return (
                  <Link
                    to="/give/history"
                    className={
                      "locked-left soft-double@lap-and-up " +
                      "soft h7 text-dark-secondary plain"
                    }
                  >
                    <i
                      className="icon-arrow-back soft-half-right display-inline-block"
                      style={{ verticalAlign: "middle" }}
                    />
                    <span
                      className="display-inline-block"
                      style={{ verticalAlign: "middle", marginTop: "5px" }}
                    >
                      Back
                    </span>
                  </Link>
                );
              }
              return null;
            })()}
            <div
              className={
                "text-left soft-double-top hard-left@lap-and-up " +
                "soft-half-bottom soft@anchored"
              }
            >
              <div className="soft-double-ends@anchored">
                {(() => {
                  const { transaction } = this.props;
                  if (!transaction) {
                    // loading
                    return (
                      <div className="text-center soft">
                        <Spinner styles={{ width: "40px", height: "40px" }} />
                      </div>
                    );
                  }
                  const { account } = transaction.details[0];
                  const { person } = transaction;
                  return (
                    <div className="text-center">
                      <p className="push-half-bottom">
                        <em>{this.formatDate(transaction.date)}</em>
                      </p>
                      <h3 className="text-dark-secondary">{account.name}</h3>

                      <Currency
                        amount={transaction.details[0].amount.toFixed(2)}
                        baseHeadingSize="1"
                        className="display-inline-block text-center soft-bottom"
                      />

                      <h6 className="push-bottom text-dark-tertiary">
                        {person.nickName || person.firstName} {person.lastName}
                      </h6>

                      {(() => {
                        const detail = transaction.payment;
                        if (detail && detail.accountNumber) {
                          return (
                            <h4 className="text-dark-secondary">
                              {detail.accountNumber.slice(-4)}&nbsp;

                              {(() => {
                                if (detail.paymentType && detail.paymentType === "ACH") {
                                  return (
                                    <AccountType width="30px" height="20px" type="Bank" />
                                  );
                                } else if (detail.paymentType) {
                                  return (
                                    <AccountType
                                      width="30px"
                                      height="20px"
                                      type={detail.paymentType}
                                    />
                                  );
                                }
                                return null;
                              })()}

                            </h4>
                          );
                        }
                        return null;
                      })()}
                      <p
                        className={
                          "text-center soft-ends soft-double@anchored " +
                          "flush-bottom soft-ends soft-sides@portable"
                        }
                      >
                        Thank you so much for your contributions! It is because of your generosity that we are able to continue telling stories of the greatness of Jesus and seeing lives changed.
                      </p>
                    </div>
                  );
                })()}
              </div>
            </div>
          </div>
          <div className="soft-half soft-sides@portable soft-double-sides@anchored">

            <h4 className="soft soft-double-ends text-center@lap-and-up flush-bottom">
              Recent Articles About Giving
            </h4>
            {(() => {
              if (loadingEntries) {
                return <div className="one-whole soft text-center"><Spinner /></div>;
              }
              return (
                <div className="grid">
                  {entries && entries.map((entry, key) => (
                    <div
                      key={key}
                      className={
                        "grid__item one-whole push-half-bottom push-bottom@portable hard-bottom"
                      }
                    >
                      <SideBySide
                        classes={["push-bottom@lap-and-up"]}
                        images={entry.content.images}
                        defaultImage={entry.content.images[0].url}
                      >
                        <h4 className="push-half-top@portable push-top@anchored">
                          {entry.title}
                        </h4>

                        <p><small dangerouslySetInnerHTML={{ __html: entry.meta.summary }} /></p>
                        {(() => {
                          if (process.env.WEB) {
                            return (
                              <a
                                rel="noopener noreferrer"
                                target="_blank"
                                href={`https://newspring.cc/articles/${entry.meta.urlTitle}`}
                                className={
                                  "h6 btn--small btn--dark-tertiary " +
                                  "soft-sides@portable one-whole@handheld"
                                }
                              >
                                Read more
                              </a>
                            );
                          }

                          if (process.env.NATIVE) {
                            return (
                              <Link
                                to={`/articles/${entry.entryId}`}
                                className={
                                  "h6 btn--small btn--dark-tertiary " +
                                  "soft-sides@portable one-whole@handheld"
                                }
                              >
                                Read more
                              </Link>
                            );
                          }

                          return null;
                        })()}


                      </SideBySide>
                    </div>
                  ))}
                </div>
              );
            })()}

          </div>
        </Left>
      </div>

    );
  }
  /* eslint-enable max-len */
}
