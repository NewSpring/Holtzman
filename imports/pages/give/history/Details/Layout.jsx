import { Component, PropTypes} from "react"
import Moment from "moment"
import { Link } from "react-router"

import Split, { Left, Right } from "../../../../blocks/split"

import { Spinner } from "../../../../components/loading"
import SideBySide from "../../../../components/cards/SideBySide"
import Meta from "../../../../components/meta"
import AccountType from "../../../../components/accountType"

export default class Layout extends Component {

  static contextTypes = {
    shouldAnimate: PropTypes.bool
  }

  formatDate = (date) => {
    return Moment(new Date(date)).format("MMM D, YYYY")
  }

  monentize = (value, fixed) => {

    if (typeof value === "number") value = `${value}`;
    if (!value.length) return `$0.00`

    value = value.replace(/[^\d.-]/g, "")

    let decimals = value.split(".")[1]
    if ((decimals && decimals.length >= 2) || fixed) {
      value = Number(value).toFixed(2)
      value = String(value)
    }

    value = value.replace(/\B(?=(\d{3})+(?!\d))/g, ",")
    return `$${value}`
  }

  render () {
    const { loadingEntries, entries } = this.props;
    return (
      <div>
        <Split nav={true} classes={["background--light-primary"]}>

          <Meta title="Giving History" />

          <Right background="//dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/all/heroes/newspring/campuses/Florence.1.2x1_1700_850_90_c1.jpg"
   mobile={false}>

          </Right>

        </Split>
        <Left scroll={true} ref="container" classes={["background--light-secondary"]}>

          <div className="soft-double-sides@lap-and-up soft-double-ends@lap-and-up soft background--light-primary">
            {(() => {
              if (process.env.WEB) {
                return (
                  <Link to="/give/history" className="locked-top locked-left soft-double@lap-and-up soft h7 text-dark-secondary plain" >
                    <i className="icon-arrow-back soft-half-right display-inline-block" style={{verticalAlign: "middle"}}></i>
                    <span className="display-inline-block" style={{verticalAlign: "middle", marginBottom: "2px"}}>Back</span>
                  </Link>
                )
              }
            })()}
            <div className="text-left soft-double-top hard-left@lap-and-up soft-half-bottom soft@anchored ">
              <div className="soft-double-ends@anchored">
                {(() => {
                  const { transaction } = this.props
                  if (!transaction) {
                    // loading
                    return (
                      <div className="text-center soft">
                        <Spinner styles={{width: "40px", height: "40px"}}/>
                      </div>
                    )
                  }
                  const { account } = transaction.details[0];
                  const { person } = transaction;
                  return (
                    <div className="text-center">
                      <p className="push-half-bottom"><em>{this.formatDate(transaction.date)}</em></p>
                      <h3 className="text-dark-secondary">{account.name}</h3>

                      <h1 className="text-dark-primary">{this.monentize(transaction.details[0].amount)}</h1>

                      <h6 className="push-bottom text-dark-tertiary">{person.nickName || person.firstName} {person.lastName}</h6>

                      {(() => {
                        const detail = transaction.payment
                        if (detail && detail.accountNumber) {
                          return (
                            <h4 className="text-dark-secondary">
                              {detail.accountNumber.slice(-4)}&nbsp;

                              {(() => {
                                if (detail.paymentType && detail.paymentType === "ACH") {
                                  return (
                                    <AccountType width="30px" height="20px" type="Bank"/>
                                  )
                                } else if (detail.paymentType) {
                                  return (
                                    <AccountType width="30px" height="20px" type={detail.paymentType} />
                                  )
                                }
                              })()}

                            </h4>
                          )
                        }
                      })()}
                      <p className="text-center soft-ends soft-double@anchored flush-bottom soft-ends soft-sides@portable">
                        Thank you so much for your contributions! It is because of your generosity we are able to continue telling stories of the greatness of Jesus and seeing peoples lives changed.
                      </p>
                    </div>
                  )
                })()}
              </div>
            </div>
          </div>
          <div className="soft-half soft-sides@portable soft-double-sides@anchored">

            <h4 className="soft soft-double-ends text-center@lap-and-up flush-bottom">
              Recent Articles About Giving
            </h4>
              {(() => {
                if (loadingEntries) (
                  <div className="one-whole soft text-center"><Spinner /></div>
                )
                return (
                  <div className="grid">
                    {entries && entries.map((entry, key) => (
                      <div key={key} className="grid__item one-whole push-half-bottom push-bottom@portable hard-bottom">
                        <SideBySide
                          classes={["push-bottom@lap-and-up"]}
                          images={entry.content.images}
                          defaultImage={entry.content.images[0].url}
                        >
                          <h4 className="push-half-top@portable push-top@anchored">
                            {entry.title}
                          </h4>

                          <p><small dangerouslySetInnerHTML={{ __html: entry.meta.summary }}></small></p>
                          {(() => {
                            if (process.env.WEB) {
                              return (
                                <a
                                  target="_blank"
                                  href={`https://newspring.cc/articles/${entry.meta.urlTitle}`}
                                  className="h6 btn--small btn--dark-tertiary soft-sides@portable one-whole@handheld"
                                >
                                  Read more
                                </a>
                              )
                            }

                            if (process.env.NATIVE) {
                              return (
                                <Link
                                  to={`/articles/${entry.entryId}`}
                                  className="h6 btn--small btn--dark-tertiary soft-sides@portable one-whole@handheld"
                                >
                                  Read more
                                </Link>
                              )
                            }
                          })()}


                        </SideBySide>
                      </div>
                    ))}
                  </div>
                )

              })()}

          </div>


        </Left>
      </div>

    )
  }
}
