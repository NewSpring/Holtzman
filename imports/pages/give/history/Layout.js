
import { Component, PropTypes } from "react";
import { Link } from "react-router";
import moment from "moment";
import Filter from "./Filter";

import { Spinner } from "../../../components/loading";
import Split, { Left, Right } from "../../../blocks/split";
import Meta from "../../../components/meta";

export function formatDate(date) {
  return moment(date).format("MMM D, YYYY");
}

export function monentize(value, fixed) {
  let strVal = typeof value === "number" ? `${value}` : value;

  if (!strVal.length) {
    return "$0.00";
  }

  strVal = strVal.replace(/[^\d.-]/g, "");

  const decimals = strVal.split(".")[1];
  if ((decimals && decimals.length >= 2) || fixed) {
    strVal = Number(strVal).toFixed(2);
    strVal = String(strVal);
  }

  strVal = strVal.replace(/\B(?=(\d{3})+(?!\d))/g, ",");
  return `$${strVal}`;
}

export const TransactionDetail = ({
  transactionDetail,
  transaction,
  icon,
  status,
  failure,
  person,
}) => (
  <div className="grid" style={{ verticalAlign: "middle" }}>
    <div className="grid__item three-fifths" style={{ verticalAlign: "middle" }}>
      <div className="relative">
        <div
          className="background--fill soft visuallyhidden@palm float-left round push-half-top"
          style={{ backgroundImage: `url("${person.photo}")` }}
        />
        <div className="soft-double-left@palm-wide-and-up push-left@palm-wide-and-up">
          <h5
            className="text-dark-secondary flush"
            style={{ textOverflow: "ellipsis", whiteSpace: "nowrap" }}
          >
            {transactionDetail.account.name}
          </h5>
          <h6 className="text-dark-tertiary soft-half-bottom flush">
            {person.firstName} {person.lastName}
          </h6>
          <p className={`flush italic small ${failure ? "text-alert" : "text-dark-tertiary"}`}>
            {status ? `${status} - ` : ""}{formatDate(transaction.date)}
          </p>
        </div>
      </div>

    </div>

    <div className="grid__item two-fifths text-right" style={{ verticalAlign: "middle" }}>
      <div className="soft-half-right">

        <h4
          className={
            "text-dark-tertiary one-whole flush " +
            "soft-right@handheld soft-double-right@lap-and-up"
          }
          style={{
            overflow: "hidden",
            textOverflow: "ellipsis",
            whiteSpace: "nowrap",
          }}
        >
          {monentize(transactionDetail.amount)}
          {(() => {
            if (icon) {
              return (
                <span
                  className="text-primary icon-arrow-next locked"
                  style={{
                    right: "-5px",
                    top: "1px",
                  }}
                />
              );
            }
            return null;
          })()}

        </h4>


      </div>

    </div>

  </div>
);

TransactionDetail.propTypes = {
  transactionDetail: PropTypes.object.isRequired,
  transaction: PropTypes.object.isRequired,
  icon: PropTypes.bool.isRequired,
  status: PropTypes.string,
  failure: PropTypes.bool,
  person: PropTypes.object.isRequired,
};

export const TransactionCard = ({ transactionDetail, transaction, person }) => {
  const { status } = transaction;

  /*

    turn on a couple pendings for UI testing

  */
  if (status && status.toLowerCase().indexOf("pending") > -1) {
    return (
      <div
        className="soft card"
        style={{
          borderStyle: "solid",
          borderColor: "f1f1f1",
          boxShadow: "none",
          borderWidth: "2px",
          backgroundColor: "transparent",
        }}
      >
        <TransactionDetail
          transactionDetail={transactionDetail}
          transaction={transaction}
          icon={false}
          status="Pending"
          person={person}
        />
      </div>
    );
  }

  /* eslint-disable max-len */
  if (status && status.toLowerCase().indexOf("failed") > -1) {
    return (
      <div
        className="soft card"
      >
        <TransactionDetail
          transactionDetail={transactionDetail}
          transaction={transaction}
          icon={false}
          status="Failed to Process"
          failure
          person={person}
        />
        <p className="flush-bottom soft-top" style={{ lineHeight: ".9" }}><small><em>
          For more information about why this contribution failed to process, please contact our Finance Team at <a href="tel:864-965-9990">864-965-9990</a> or <a rel="noopener noreferrer" target="_blank" href="//rock.newspring.cc/workflows/152?Topic=Stewardship">contact us</a>
        </em></small></p>
      </div>
    );
  }
  /* eslint-enable max-len */
  return (
    <div className="soft card">
      <Link to={`/give/history/${transaction.id}`}>
        <TransactionDetail
          transactionDetail={transactionDetail}
          transaction={transaction}
          icon
          person={person}
        />
      </Link>
    </div>
  );
};

TransactionCard.propTypes = {
  transactionDetail: PropTypes.object.isRequired,
  transaction: PropTypes.object.isRequired,
  person: PropTypes.object.isRequired,
};

export default class Layout extends Component {

  static propTypes = {
    transactions: PropTypes.array.isRequired,
    ready: PropTypes.bool,
    Loading: PropTypes.func.isRequired,
    done: PropTypes.bool,
    changeFamily: PropTypes.func.isRequired,
    changeDates: PropTypes.func.isRequired,
    reloading: PropTypes.bool,
    family: PropTypes.array.isRequired,
  }

  monentize = monentize
  formatDate = formatDate

  render() {
    const {
      transactions,
      ready,
      Loading,
      done,
      changeFamily,
      changeDates,
      reloading,
    } = this.props;


    /* eslint-disable max-len */
    return (

      <div>
        <Split nav classes={["background--light-primary"]}>

          <Meta title="Giving History" />

          <Right
            background="//dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/newspring/_fpo/NScollege-cip-0033_1700_1133_90_c1.jpg"
            mobile={false}
          />


        </Split>
        <Left scroll ref="container" classes={["background--light-secondary"]}>


          <div
            className="soft-double-sides@lap-and-up soft-ends@lap-and-up background--light-primary"
          >
            <div className="soft soft-double-ends hard-left@lap-and-up">
              <h2 className="flush hard">Giving History</h2>
            </div>
          </div>

          <Filter
            family={this.props.family}
            changeFamily={changeFamily}
            changeDates={changeDates}
          />
          <div
            className={
              "soft-half soft@portable soft-double@anchored " +
              "soft-double-bottom@anchored soft-bottom@portable"
            }
            ref="history"
          >
            {(() => {
              if (reloading || (!transactions.length && !ready)) {
                // loading
                return (
                  <div className="text-center soft">
                    <Spinner styles={{ width: "40px", height: "40px" }} />
                  </div>

                );
              } else if (!transactions.length && ready) {
                return (
                  <div className="text-left soft-ends soft-half-sides">
                    <p>
                      We didn&#39;t find any contributions associated with your account. If you would like to start giving, click <Link to="/give/now">here</Link>
                    </p>
                    <p><em>If you have any questions, please call our Finance Team at 864-965-9990 or <a rel="noopener noreferrer" target="_blank" href="//rock.newspring.cc/workflows/152?Topic=Stewardship">contact us </a> and someone will be happy to assist you.</em></p>
                  </div>
                );
              }

              let lastYear = null;
              return (
                <div>
                  {transactions.map((transaction, key) => {
                    const { details, person } = transaction;
                    return (
                      <div key={key}>
                        {details.map((transactionDetail, i) => {
                          if (!transactionDetail.account) return null;

                          if (Number(transactionDetail.amount) === 0) return null;

                          const year = moment(transaction.date).year();
                          if (year !== lastYear) {
                            lastYear = year;
                            return (
                              <div key={i}>
                                <div className="soft soft-half-left text-left">
                                  <h5>{year}</h5>
                                </div>
                                <TransactionCard
                                  transaction={transaction}
                                  transactionDetail={transactionDetail}
                                  person={person}
                                />
                              </div>
                            );
                          }

                          return (
                            <TransactionCard
                              transaction={transaction}
                              transactionDetail={transactionDetail}
                              person={person}
                              key={i}
                            />
                          );
                        })}
                      </div>

                    );
                  })}
                </div>
              );
            })()}
          </div>

          {/* Load more */}
          <div className="one-whole text-center">
            {(() => {
              if (!ready && !transactions.length) return null;
              if (!transactions.length && ready) return null;
              if (reloading) return null;

              if (done) return <p><small><em>No More Contributions</em></small></p>;
              return <Loading />;
            })()}
          </div>


        </Left>
      </div>
    );
    /* eslint-enable max-len */
  }
}
