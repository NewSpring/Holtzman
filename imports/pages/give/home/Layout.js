// @flow

import Activity from "./Activity";
import Schedules from "./Schedules";
import SavedPayments from "./SavedPayments";

const Layout = () => (
  <div>
    <Activity />
    <Schedules />
    <SavedPayments />
  </div>
);

export default Layout;
