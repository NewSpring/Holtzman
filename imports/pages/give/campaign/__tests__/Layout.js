/* eslint-disable */
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Layout from "../Layout";
// import { reset, startBuffering } from "aphrodite/lib/inject";

const account1 = [
  {
    description: "Web Team Tesla Fund",
    name: "Tesla Fund",
    id: "e450b1f1dbb79ffea48d9aba1a00ea0f2885d4bfe435683d1b1129fae408729d",
    summary: "Tesla, because they're awesome!",
    image: "https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png",
    order: 0,
    images: {
      fileName: "Image%202016-11-07%20at%202.38.14%20PM.png",
      fileType: null,
      fileLabel: "2:1",
      s3: "https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png",
      cloudfront: "https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png",
    },
  },
];

const account2 = [
  {
    description: "Web Team Tesla Fund",
    name: "Tesla Fund",
    id: "e450b1f1dbb79ffea48d9aba1a00ea0f2885d4bfe435683d1b1129fae408729d",
    summary: "Tesla, because they're awesome!",
    image: "https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png",
    order: 0,
  },
];

// XXX probably will need to switch back to create-react-renderer
describe("Template", () => {
  it("looks empty with no props", () => {
    const tree = shallow(
      <Layout
        account={""}
      />
    );
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it("accepts account object with complete data", () => {
    const tree = shallow(
      <Layout
        account={account1}
      />
    );
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it("accepts account object without image data", () => {
    const tree = shallow(
      <Layout
        account={account2}
      />
    );
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

