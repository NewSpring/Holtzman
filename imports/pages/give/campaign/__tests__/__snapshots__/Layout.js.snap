// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template accepts account object with complete data 1`] = `
<div>
  <Connect(SplitContainerWithoutData)
    classes={
      Array [
        "background--light-primary",
      ]
    }
    nav={true}
  >
    <Meta
      description="Tesla, because they're awesome!"
      image="https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png"
      meta={
        Array [
          Object {
            "content": "article",
            "property": "og:type",
          },
        ]
      }
      title="Tesla Fund"
    />
    <Right
      background="https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png"
      mobile={true}
    />
    <Right
      background="https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png"
      mobile={false}
    />
  </Connect(SplitContainerWithoutData)>
  <Left
    classes={
      Array [
        "background--light-primary",
      ]
    }
    scroll={true}
  >
    <Link
      className="locked-top locked-left soft-double@lap-and-up soft h7 text-dark-secondary plain visuallyhidden@handheld"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/give/now"
    >
      <i
        className="icon-arrow-back soft-half-right display-inline-block"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }
      />
      <span
        className="display-inline-block"
        style={
          Object {
            "marginTop": "5px",
            "verticalAlign": "middle",
          }
        }
      >
        Back
      </span>
    </Link>
    <div
      className="soft@lap-and-up soft-double-top@lap-and-up"
    >
      <div
        className="soft soft-double-bottom soft-double-top@lap-and-up"
      >
        <h2>
          Tesla Fund
        </h2>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Web Team Tesla Fund",
            }
          }
        />
      </div>
    </div>
    <div
      className="background--light-secondary"
    >
      <div
        className="constrain-copy soft-double@lap-and-up"
      >
        <div
          className="soft soft-double-bottom soft-double-top@lap-and-up"
        >
          <Connect(CartContainer)
            accounts={
              Array [
                Object {
                  "description": "Web Team Tesla Fund",
                  "id": "e450b1f1dbb79ffea48d9aba1a00ea0f2885d4bfe435683d1b1129fae408729d",
                  "image": "https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png",
                  "images": Array [
                    Object {
                      "cloudfront": "https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png",
                      "fileLabel": "2:1",
                      "fileName": "Image%202016-11-07%20at%202.38.14%20PM.png",
                      "fileType": null,
                      "s3": "https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png",
                    },
                  ],
                  "name": "Tesla Fund",
                  "order": 0,
                  "summary": "Tesla, because they're awesome!",
                },
              ]
            }
            donate={true}
          />
        </div>
      </div>
    </div>
  </Left>
</div>
`;

exports[`Template accepts account object without image data 1`] = `
<div>
  <Connect(SplitContainerWithoutData)
    classes={
      Array [
        "background--light-primary",
      ]
    }
    nav={true}
  >
    <Meta
      description="Tesla, because they're awesome!"
      image="https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png"
      meta={
        Array [
          Object {
            "content": "article",
            "property": "og:type",
          },
        ]
      }
      title="Tesla Fund"
    />
    <Right
      background="https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png"
      mobile={true}
    />
    <Right
      background="https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png"
      mobile={false}
    />
  </Connect(SplitContainerWithoutData)>
  <Left
    classes={
      Array [
        "background--light-primary",
      ]
    }
    scroll={true}
  >
    <Link
      className="locked-top locked-left soft-double@lap-and-up soft h7 text-dark-secondary plain visuallyhidden@handheld"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/give/now"
    >
      <i
        className="icon-arrow-back soft-half-right display-inline-block"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }
      />
      <span
        className="display-inline-block"
        style={
          Object {
            "marginTop": "5px",
            "verticalAlign": "middle",
          }
        }
      >
        Back
      </span>
    </Link>
    <div
      className="soft@lap-and-up soft-double-top@lap-and-up"
    >
      <div
        className="soft soft-double-bottom soft-double-top@lap-and-up"
      >
        <h2>
          Tesla Fund
        </h2>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Web Team Tesla Fund",
            }
          }
        />
      </div>
    </div>
    <div
      className="background--light-secondary"
    >
      <div
        className="constrain-copy soft-double@lap-and-up"
      >
        <div
          className="soft soft-double-bottom soft-double-top@lap-and-up"
        >
          <Connect(CartContainer)
            accounts={
              Array [
                Object {
                  "description": "Web Team Tesla Fund",
                  "id": "e450b1f1dbb79ffea48d9aba1a00ea0f2885d4bfe435683d1b1129fae408729d",
                  "image": "https://cl.ly/i26e/Image%202016-11-07%20at%202.38.14%20PM.png",
                  "name": "Tesla Fund",
                  "order": 0,
                  "summary": "Tesla, because they're awesome!",
                },
              ]
            }
            donate={true}
          />
        </div>
      </div>
    </div>
  </Left>
</div>
`;

exports[`Template looks empty with no props 1`] = `
<div>
  <Connect(SplitContainerWithoutData)
    classes={
      Array [
        "background--light-primary",
      ]
    }
    nav={true}
  >
    <Meta
      meta={
        Array [
          Object {
            "content": "article",
            "property": "og:type",
          },
        ]
      }
    />
    <Right
      mobile={true}
    />
    <Right
      mobile={false}
    />
  </Connect(SplitContainerWithoutData)>
  <Left
    classes={
      Array [
        "background--light-primary",
      ]
    }
    scroll={true}
  >
    <Link
      className="locked-top locked-left soft-double@lap-and-up soft h7 text-dark-secondary plain visuallyhidden@handheld"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/give/now"
    >
      <i
        className="icon-arrow-back soft-half-right display-inline-block"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }
      />
      <span
        className="display-inline-block"
        style={
          Object {
            "marginTop": "5px",
            "verticalAlign": "middle",
          }
        }
      >
        Back
      </span>
    </Link>
    <div
      className="soft@lap-and-up soft-double-top@lap-and-up"
    >
      <div
        className="soft soft-double-bottom soft-double-top@lap-and-up"
      >
        <h2 />
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
      </div>
    </div>
    <div
      className="background--light-secondary"
    >
      <div
        className="constrain-copy soft-double@lap-and-up"
      >
        <div
          className="soft soft-double-bottom soft-double-top@lap-and-up"
        >
          <Connect(CartContainer)
            accounts={
              Array [
                "",
              ]
            }
            donate={true}
          />
        </div>
      </div>
    </div>
  </Left>
</div>
`;
